SceneExplorer/ToBeReplaced/Helpers/UIStyle.cs

```diff
@@ -7,11 +7,20 @@ namespace SceneExplorer.ToBeReplaced.Helpers
 {
     public class UIStyle
     {
+        private static UIStyle _instance;
+
         public static UIStyle Instance
         {
             [MethodImpl(MethodImplOptions.NoInlining)]
-            get;
-            set;
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new UIStyle();
```

Comment:
How are you ensuring this constructor will always be invoked from `OnGUI` method?
`GUI` calls have to be called from inside OnGUI method, that's why it was done such strange way (first access of UIStyle instace was creating it)

Reply:
Well, how do you ensure that with the Build method? I learned this fact about it the hard way, calling the Build method from outside of OnGui :D
I have two alternatives to mitigate this:
- Add some warnings to the doc of the `Instance` property about it must be called from OnGui
- Or refactor it again, and instead of a Singletone use locally created `UIStyle` instances in every class. It would make the code somewhat cleaner, but it also creates some overhead, creating multiple `UIStyle` instances.
Any other suggesstions?

Reply:
Build was called when FloatingWindow instance was called by Unity for the first time (from OnGUI), caching style as static style for all floating windows (and inside UIStyle). the rest of the code (ui helpers and controls) were using UIStyle instance but since they are helpers the would be invoked only by some call coming from OnGUI - style already set. If you make mistake and try to use UIStyle before it was instantiated you will be greeted with `NullReferenceException` at runtime. 
I'm more or less halfway through rewriting UI so all this code will be removed soon

Reply:
To address this I made the following changes:
- Restored the `UIStyle.Build()` method and ensured that only it can create the `UIStyle` instance (by setting the `UIStyle ctor` to private).
- I also restored the call to `UIStyle.Build()` from `FloatingWindowBase.OnGUI()` and made sure that it is called only once with a `static bool _uiStyleWasCreated` flag.
- I did not restore the `FloatingWindowBase.Style` property because there were already mixed usages in `FloatingWindowBase` and in other classes inherited from it: Sometimes the local `Style` property was used, sometimes the `UIStyle.Instance` property (e.g. in `CalculateBackgroundStyle`). I wanted to make sure that there is only one access point for UIStyle.
- Theoretically any `OnGUI` method could be potentially the first to create `UIStyle`, but in practice it will always be `FloatingWindowBase.OnGUI`, so I did not bother adding the `UIStyle.Build()` call to `DropdownControl.OnGUI()`...

I suppose now it should be fine until you finish the new UI :)